src/psyclone/psyGen.py

```diff
@@ -2749,6 +2735,44 @@ def coloured_text(self):
 
 
 class Loop(Node):
+    '''Represents a loop in the PSyIR.
+
+    :param parent: Parent of this node in the PSyIR.
+    :type parent: sub-class of :py:class:`psyclone.psyGen.Node`
+    :param str variable_name: Optional name of the loop iterator \
+    variable. Defaults to an empty string.
+    :param valid_loop_types: A list of loop types that are specific \
+    to a particular API.
+    :type valid_loop_types: list of str
+
+    '''
+
+    def __init__(self, parent=None,
+                 variable_name="",
+                 valid_loop_types=[]):
```

Comment:
`pylint` says: `W:2750, 4: Dangerous default value [] as argument (dangerous-default-value)`, however I am not sure what would be a smarter way to initialise a list so I guess we can ignore this.

Reply:
Modified to default to None.